cloud/linode/fake_linode_test.go

```diff
@@ -72,346 +75,203 @@ func (f *fakeAPI) didRequestOccur(method, path, body string) bool {
+		res := 0
+		data := []linodego.NodeBalancer{}
+		filter := r.Header.Get("X-Filter")
+		if filter == "" {
+			for _, n := range f.nb {
+				data = append(data, *n)
+			}
+		} else {
+			var fs map[string]string
+			err := json.Unmarshal([]byte(filter), &fs)
+			if err != nil {
+				f.t.Fatal(err)
 			}
-			rx, _ = regexp.Compile("/nodebalancers/[0-9]+/configs/[0-9]+")
-			if rx.MatchString(urlPath) {
-				id := filepath.Base(urlPath)
-				nbc, found := f.nbc[id]
-				if found {
-					rr, _ := json.Marshal(nbc)
-					_, _ = w.Write(rr)
-
-				} else {
-					w.WriteHeader(404)
-					resp := linodego.APIError{
-						Errors: []linodego.APIErrorReason{
-							{Reason: "Not Found"},
-						},
-					}
-					rr, _ := json.Marshal(resp)
-					_, _ = w.Write(rr)
+			for _, n := range f.nb {
+				if (n.Label != nil && fs["label"] != "" && *n.Label == fs["label"]) ||
+					(fs["ipv4"] != "" && n.IPv4 != nil && *n.IPv4 == fs["ipv4"]) {
+					data = append(data, *n)
 				}
-				return
 			}
-			rx, _ = regexp.Compile("/nodebalancers/[0-9]+/configs")
-			if rx.MatchString(urlPath) {
-				res := 0
-				data := []linodego.NodeBalancerConfig{}
-				filter := r.Header.Get("X-Filter")
-				if filter == "" {
-					for _, n := range f.nbc {
-						data = append(data, *n)
-					}
-				} else {
-					var fs map[string]string
-					err := json.Unmarshal([]byte(filter), &fs)
-					if err != nil {
-						f.t.Fatal(err)
-					}
-					for _, n := range f.nbc {
-						if strconv.Itoa(n.NodeBalancerID) == fs["nodebalancer_id"] {
-							data = append(data, *n)
-						}
-					}
-				}
-				resp := linodego.NodeBalancerConfigsPagedResponse{
-					PageOptions: &linodego.PageOptions{
-						Page:    1,
-						Pages:   1,
-						Results: res,
-					},
-					Data: data,
-				}
-				rr, err := json.Marshal(resp)
-				if err != nil {
-					f.t.Fatal(err)
-				}
-				_, _ = w.Write(rr)
-				return
+		}
+		resp := linodego.NodeBalancersPagedResponse{
+			PageOptions: &linodego.PageOptions{
+				Page:    1,
+				Pages:   1,
+				Results: res,
+			},
+			Data: data,
+		}
+		rr, _ := json.Marshal(resp)
+		_, _ = w.Write(rr)
+	})
+
+	f.mux.HandleFunc("GET /v4/nodebalancers/{nodeBalancerId}", func(w http.ResponseWriter, r *http.Request) {
+		nb, found := f.nb[r.PathValue("nodeBalancerId")]
+		if found {
+			rr, _ := json.Marshal(nb)
+			_, _ = w.Write(rr)
+
+		} else {
+			w.WriteHeader(404)
+			resp := linodego.APIError{
+				Errors: []linodego.APIErrorReason{
+					{Reason: "Not Found"},
+				},
 			}
+			rr, _ := json.Marshal(resp)
+			_, _ = w.Write(rr)
+		}
+	})
 
-			rx = regexp.MustCompile("/nodebalancers/[0-9]+/firewalls")
-			if rx.MatchString(urlPath) {
-				id := strings.Split(urlPath, "/")[2]
-				devID, err := strconv.Atoi(id)
-				if err != nil {
-					f.t.Fatal(err)
-				}
+	f.mux.HandleFunc("GET /v4/nodebalancers/{nodeBalancerId}/firewalls", func(w http.ResponseWriter, r *http.Request) {
+		devID, err := strconv.Atoi(r.PathValue("nodeBalancerId"))
```

Comment:
```suggestion
		nodeBalancerId, err := strconv.Atoi(r.PathValue("nodeBalancerId"))
```

nit: The f.fwd field is a bit of a head scratcher so using `nodeBalancerID` keeps the variable purpose front and center.